core/primitives/src/epoch_manager.rs

```diff
@@ -83,11 +83,17 @@ pub struct AllEpochConfig {
     use_production_config: bool,
     /// EpochConfig from genesis
     genesis_epoch_config: EpochConfig,
+    /// Chain Id. Some parameters are specific to certain chains.
+    chain_id: String,
 }
 
 impl AllEpochConfig {
-    pub fn new(use_production_config: bool, genesis_epoch_config: EpochConfig) -> Self {
-        Self { use_production_config, genesis_epoch_config }
+    pub fn new(
+        use_production_config: bool,
+        genesis_epoch_config: EpochConfig,
+        chain_id: &str,
+    ) -> Self {
+        Self { use_production_config, genesis_epoch_config, chain_id: chain_id.to_string() }
     }
 
     pub fn for_protocol_version(&self, protocol_version: ProtocolVersion) -> EpochConfig {
```

Comment:
When testing on mocknet please remember to set `use_production_config` to true, otherwise your logic won't be triggered - see two lines below.

Reply:
Above you suggested to remove `use_production_config`. That should take care of mocknet configuration

Reply:
Sure, as long as you do that first, you'll be fine. It will make my life easier as well when testing resharding.